airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,148 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
                     {properties}"""
             )
 
+    def test_property_type_is_not_array(self, connector_spec_dict: dict):
+        """
+        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, value in dpath.util.search(specification, "**/properties/*/type", yielded=True):
```

Comment:
```suggestion
        for type_path, type_value in dpath.util.search(specification, "**/properties/*/type", yielded=True):
```
nit: for consistency